mne/io/eeglab/eeglab.py

```diff
@@ -24,6 +30,64 @@
 CAL = 1e-6
 
 
+def _todict_from_np_struct(data):  # taken from pymatreader.utils
```

Comment:
I think we normally put vendored code into `mne.fixes`?  And then use a... shim? (not sure the right term) to call the function from pymatreader if the user has it installed, and call it from our vendored code if they don't.  (I think the rationale is that users who have pymatreader installed can benefit from upstream changes even if we fail to re-vendor frequently).

Reply:
But this does not really *fix* anything, so I'm not sure it fits. We used to vendor packages in `mne.externals`, but we moved away from that and instead require those packages directly.

The problem is that our EEGLAB code is tailored towards the output of `pymatreader`, which basically means that it expects a dict where values are lists, but `scipy.io.loadmat()` produces a list of dicts instead.

Reply:
> But this does not really _fix_ anything, so I'm not sure it fits.

🤷🏻 yeah, I guess I see your point, what's in `mne.fixes.py` is (from what I can tell from code comments) copies of stuff from other libraries *that we've tweaked in some way* (either because upstream won't fix, or because we want to make sure users with older versions of dependencies benefit from the fix no matter what).

Reply:
Maybe putting those functions, together with a unified loader, into e.g. `mne/io/eeglab/_eeglab.py` would be a good idea then?

Reply:
sounds good.